syntax = "proto3";

option go_package = "github.com/pkulik0/bufro/api/internal/pb";

enum BufType {
    BUF_TYPE_UNKNOWN = 0;
    BUF_TYPE_IMAGE = 1;
    BUF_TYPE_VIDEO = 2;
}

message CreateBufRequest {
    BufType type = 1;
    bytes data = 2;
}

message CreateBufResponse {
    string id = 1;
}

message Error {
    string message = 1;
}

message GetUserBufsRequest {
    string user_id = 1;
    int64 limit = 2;
    int64 offset = 3;
}

message Buf {
    string id = 1;
    string user_id = 2;
    BufType type = 3;
    bytes data = 4;
}

message GetUserBufsResponse {
    repeated Buf bufs = 1;
}